R/man.R

```diff
@@ -73,65 +16,59 @@
     }
   }
 
+  n <- length(man_source)
+
+  # can't use message_info with {}
+  cli::cli_alert_info("Found {n} man page{?s} to convert.")
+  i <- 0
+  cli::cli_progress_step("Converting {cli::qty(n)}man page{?s}: {i}/{n}", spinner = TRUE)
+
+  conversion_worked <- vector(length = n)
+
   # process man pages one by one
-  for (f in man_source) {
-    origin_Rd <- fs::path_join(c("man", fs::path_ext_set(f, ".Rd")))
+  for (i in seq_along(man_source)) {
+    f <- man_source[i]
+    # fs::path_ext_set breaks filenames with dots, ex: 'foo.bar.Rd'
+    origin_Rd <- fs::path_join(c("man", paste0(f, ".Rd")))
     destination_dir <- fs::path_join(c(.doc_path(path = "."), "man"))
-    destination_qmd <- fs::path_join(c(destination_dir, fs::path_ext_set(f, ".qmd")))
-    destination_md <- fs::path_join(c(destination_dir, fs::path_ext_set(f, ".md")))
+    destination_qmd <- fs::path_join(c(destination_dir, paste0(f, ".qmd")))
+    destination_md <- fs::path_join(c(destination_dir, paste0(f, ".md")))
     fs::dir_create(destination_dir)
     .rd2qmd(origin_Rd, destination_dir)
-    .qmd2md(destination_qmd, destination_dir)
+    conversion_worked[i] <- .qmd2md(destination_qmd, destination_dir, verbose = verbose)
     fs::file_delete(destination_qmd)
-  }
-
-  cli::cli_alert_success("Functions reference updated in `docs/man/` directory.")
-}
 
+    # section headings are too deeply nested by default
+    # this is a hack because it may remove one # from comments. But that's
+    # probably not the end of the world, because the line stick stays commented
+    # out.
+    if (fs::file_exists(destination_md)) {
+      tmp <- readLines(destination_md)
+      tmp <- gsub("^##", "#", tmp)
+      writeLines(tmp, destination_md)
+    }
```

Comment:
I think a more robust way to handle this is what I've done here: https://github.com/etiennebacher/altdoc/blob/22896e6a8e20087202c93dcba9a2a0054a63fa69/R/reformat.R#L66-L89

Reply:
I'll look into this when I tackle dependencies. https://github.com/etiennebacher/altdoc/issues/88